enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -566,22 +591,43 @@ def can_redeem(self, request, enterprise_customer_uuid):
                         "reason": reason,
                         "policy_uuids": [policy.uuid for policy in policies],
                     })
+
+            # Select one redeemable policy (i.e. use the "policy resolver") and serialize it for output.
+            resolved_policy = None
             if redeemable_policies:
                 resolved_policy = SubsidyAccessPolicy.resolve_policy(redeemable_policies)
-                serialized_policy = serializers.SubsidyAccessPolicyRedeemableResponseSerializer(resolved_policy).data
 
+            # Determine if the learner has already redeemed the requested content_key.
             has_successful_redemption = any(
                 redemption['state'] == TransactionStateChoices.COMMITTED
                 for redemption in redemptions
             )
-            can_redeem_for_content_response = {
+
+            # Determine the price for content for display purposes only.
+            try:
+                content_metadata = self.subsidy_client.get_subsidy_content_data(enterprise_customer_uuid, content_key)
+                # Note that the "content_price" key is guaranteed to exist, but the value may be None.  Also, the
+                # upstream field type is a string representing a decimal dollar.
+                list_price_decimal_dollars_str = content_metadata["content_price"]
+                list_price_integer_cents = int(float(list_price_decimal_dollars_str) * 100)
```

Comment:
If `list_price_decimal_dollars_str` can be `None`, won't an exception be raised here: `float(list_price_decimal_dollars_str)`?

```python
>>> float(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: float() argument must be a string or a number, not 'NoneType'
```

The try/except doesn't catch `TypeError` though, so I believe this exception would be raised.

Reply:
handled case, and added new test.